server/config/controller.ts

```diff
@@ -3,48 +3,65 @@
  * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
  */
 
+import merge from 'lodash.merge';
 import { jsonToSchema } from '@walmartlabs/json-to-simple-graphql-schema/lib';
-import { featureFlags, client, server } from 'ui-config';
+import { IResolvers } from 'apollo-server-express';
+import { serverConfigType } from 'types';
+import { featureFlags, client, server } from 'ui-config/index';
 
-// iterate through the various imported config items, including only items which are not sensitive
-const configToHost = {
-  featureFlags: featureFlags.publicValues,
-  client: client.publicValues,
-  server: server.publicValues,
+type configSchemaResolverType = {
+  typeDefs: string;
+  resolvers: IResolvers;
 };
 
-const { query, schemaTypes, resolvers } = Object.entries(configToHost)
-  .map(([key, config]) => ({
-    type: key,
-    typeValue: { _generatedTypeName: key, ...config },
-  }))
-  .reduce(
-    ({ schemaTypes, resolvers }, { type, typeValue }) => {
-      const { value } = jsonToSchema({
-        baseType: type,
-        jsonInput: JSON.stringify(typeValue),
-      });
-      const allResolvers = { ...resolvers.Query, [type]: () => typeValue };
+export const returnSchemaAndResolversGivenConfig: (
+  config: serverConfigType
+) => configSchemaResolverType = (config) => {
+  const { featureFlags: featureFlagOverrides } = config;
+  const { configOverrides } = config.client;
 
-      return {
-        schemaTypes: schemaTypes.concat([value]),
-        resolvers: {
-          Query: {
-            ...allResolvers,
+  const configToHost = {
+    featureFlags: merge(
+      merge({}, featureFlags.publicValues),
```

Comment:
Why the extra `merge`? `merge` is variadic so you can do `merge({}, x, y, z)`

Reply:
Aha! Much nicer! Will update (along with others)